packages/example/src/1week/URL/addQuery/__submit__/minsoo-web.ts

```diff
@@ -0,0 +1,65 @@
+const hasIncludes = (url: string, value: string): 'YES' | 'NO' =>
+  url.includes(value) ? 'YES' : 'NO'
+
+export const splitHashPart = (
+  text: string,
+): { nonHash: string; hash: string } => {
+  const indexOfHash = text.indexOf('#')
+
+  if (indexOfHash >= 0) {
+    return {
+      nonHash: text.slice(0, indexOfHash),
+      hash: text.slice(indexOfHash),
+    }
+  }
+
+  return {
+    nonHash: text,
+    hash: '',
+  }
+}
+
+export const replaceKeyValue = (
+  url: string,
+  key: string,
+  newValue: string | number,
+) => {
+  const regExp = new RegExp(`${key}=([^&]*)`, 'g')
+  if (url.match(regExp)) {
+    return url.replace(regExp, `${key}=${newValue}`)
+  }
+
+  return url
+}
+
+/**
+ * 출제자: 예진님 (yejineee)
+ * URL의 쿼리스트링 관련 유틸 함수를 계산 을 최대한 이용하여 만들기
+ *
+ * - 해쉬가 있다면, 유지해야 함
+ * - 기존 쿼리가 있다면, 유지해야 함
+ * - 기존 쿼리와 추가하려는 쿼리의 key가 동일하다면, 기존 쿼리가 대체됨
+ */
+
+export const addQuery = (
+  originURL: string,
+  key: string,
+  value: string | number,
+) => {
```

Comment:
> - 해쉬가 있다면, 유지해야 함
> - 기존 쿼리가 있다면, 유지해야 함
> - 기존 쿼리와 추가하려는 쿼리의 key가 동일하다면, 기존 쿼리가 대체됨

위와 같은 요구사항이라면 set이라는 이름을 쓰면 어떨까요? 뭔가 set이 더 어울리는 것 같아요
addQuery라면 queryParam을 배열로 사용하는 경우로 오해받을 수도 있을 거 같아보여서요

```suggestion
/**
 * 출제자: 예진님 (yejineee)
 * URL의 쿼리스트링 관련 유틸 함수를 계산 을 최대한 이용하여 만들기
 *
 * - 해쉬가 있다면, 유지해야 함
 * - 기존 쿼리가 있다면, 유지해야 함
 * - 기존 쿼리와 추가하려는 쿼리의 key가 동일하다면, 기존 쿼리가 대체됨
 */

export const setQuery = (
  originURL: string,
  key: string,
  value: string | number,
) => {
```

Reply:
오,, 저두 이거 약간 헷갈리는 것 같긴했는데 출제자인 예진님이 함수 이름을 정해주신거라! 예진님 의견도 궁금합니다!